packages/amplify-e2e-tests/src/__tests__/auth/admin-api.test.ts

```diff
@@ -0,0 +1,177 @@
+    expect(responseBody.Groups.length).toBe(2);
+  });
+
+  it('can list groups with limit', async () => {
+    const request = {
+      path: '/listGroups',
+      httpMethod: 'GET',
+      queryStringParameters: {
+        limit: '1',
+      },
+    };
+    const response = await callAdminLambda(request);
+    expect(response.statusCode).toBe(200);
+    const responseBody = JSON.parse(response.body);
+    expect(responseBody.Groups.length).toBe(1);
+  });
+
+  it('can list users', async () => {
+    const request = {
+      path: '/listUsers',
+      httpMethod: 'GET',
+    };
+    const response = await callAdminLambda(request);
+    expect(response.statusCode).toBe(200);
+    const responseBody = JSON.parse(response.body);
+    expect(responseBody.Users.length).toBe(10);
+  });
+
+  it('can list users with limit', async () => {
+    const request = {
+      path: '/listUsers',
+      httpMethod: 'GET',
+      queryStringParameters: {
+        limit: '1',
+      },
+    };
+    const response = await callAdminLambda(request);
+    expect(response.statusCode).toBe(200);
+    const responseBody = JSON.parse(response.body);
+    expect(responseBody.Users.length).toBe(1);
+  });
+
+  it('can list users in group', async () => {
+    const request = {
+      path: '/listUsersInGroup',
+      httpMethod: 'GET',
+      queryStringParameters: {
+        groupname: 'Admins',
+      },
+    };
+    const response = await callAdminLambda(request);
+    expect(response.statusCode).toBe(200);
+    const responseBody = JSON.parse(response.body);
+    expect(responseBody.Users.length).toBe(5);
+  });
+
+  it('can list users in group with limit', async () => {
+    const request = {
+      path: '/listUsersInGroup',
+      httpMethod: 'GET',
+      queryStringParameters: {
+        limit: '1',
+        groupname: 'Admins',
+      },
+    };
+    const response = await callAdminLambda(request);
+    expect(response.statusCode).toBe(200);
+    const responseBody = JSON.parse(response.body);
+    expect(responseBody.Users.length).toBe(1);
+  });
+
+  it('can list groups for user', async () => {
+    const request = {
+      path: '/listGroupsForUser',
+      httpMethod: 'GET',
+      queryStringParameters: {
+        username: 'testAdminUser1',
+      },
+    };
+    const response = await callAdminLambda(request);
+    expect(response.statusCode).toBe(200);
+    const responseBody = JSON.parse(response.body);
+    expect(responseBody.Groups.length).toBe(2);
+  });
+
+  it('can list groups for user with limit', async () => {
+    const request = {
+      path: '/listGroupsForUser',
+      httpMethod: 'GET',
+      queryStringParameters: {
+        limit: '1',
+        username: 'testAdminUser1',
+      },
+    };
+    const response = await callAdminLambda(request);
+    expect(response.statusCode).toBe(200);
+    const responseBody = JSON.parse(response.body);
+    expect(responseBody.Groups.length).toBe(1);
+  });
+});
```

Comment:
Do we need to test a failure case? e.g. the error in the description

Reply:
Before applying fix (changes in `packages/amplify-category-auth/resources/adminAuth/admin-auth-app.js` file) the tests that explicitly specify `limit` would fail with 500 error. (which I locally verified before applying the fix).
After fixing the implementation (adding `parseInt`) they're now passing.